### PR TITLE
test(ec2Query): add ec2QueryName, xmlName distinction edge case

### PR DESCRIPTION
**Pull request by siddsriv**
#### Background
add test for an edge case where `ec2QueryName`, `xmlName`, default member name being distinct caused a silent serialization bug for the AWS SDK JSv3.

#### Testing
built locally and tested with JSv3/smithy-typescript.  New test case `Ec2QueryNameDistinctFromXmlNameAndMemberName` generated and passes.

#### Links
Internal JS-6663
also, see: https://github.com/aws/aws-sdk-js-v3/pull/7734

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-protocol-tests/model/ec2Query/input.smithy [hunk 1]
```diff
@@ -198,6 +198,27 @@ apply SimpleInputParams @httpRequestTests([
...
+        protocol: ec2Query,
+        method: "POST",
+        uri: "/",
+        headers: {
+            "Content-Type": "application/x-www-form-urlencoded"
+        },
+        requireHeaders: [
+            "Content-Length"
+        ],
+        body: "Action=SimpleInputParams&Version=2020-01-08&DistinctQueryName=value1&DistinctQueryAndXmlName=value2&DistinctXmlName=value3",
```
Comment (kuhe) [inline smithy-aws-protocol-tests/model/ec2Query/input.smithy]:
```suggestion
        body: "Action=SimpleInputParams&Version=2020-01-08&QueryName=value1&QueryAndXmlName=value2&XmlNameOnly=value3",
```

the body would use the intended serialization values from the traits

File: smithy-aws-protocol-tests/model/ec2Query/input.smithy [hunk 2]
```diff
@@ -276,6 +297,18 @@ structure SimpleInputParamsInput {
 
     @xmlName("usesXmlName")
     UsesXmlName: String,
+
+    // Edge case: memberName, xmlName, and ec2QueryName all distinct
+    @ec2QueryName("QueryName")
```
Comment (kuhe) [inline smithy-aws-protocol-tests/model/ec2Query/input.smithy:303]: these members could use a documentation tag or comment to explain the desired serialization inline

```
DistinctQueryName -> QueryName
DistinctQueryAndXmlName -> QueryAndXmlName
DistinctXmlName -> XmlNameOnly
```

File: smithy-aws-protocol-tests/model/ec2Query/input.smithy [hunk 3]
```diff
@@ -276,6 +297,21 @@ structure SimpleInputParamsInput {
...
+    DistinctQueryName: String,
+
+    // Serializes as "QueryAndXmlName" (from @ec2QueryName trait, takes precedence over @xmlName)
+    @ec2QueryName("QueryAndXmlName")
+    @xmlName("xmlAndQueryName")
+    DistinctQueryAndXmlName: String,
+
+    // Serializes as "XmlNameOnly" (from @xmlName trait, no @ec2QueryName specified)
+    @xmlName("XmlNameOnly")
```
Comment (kuhe) [inline smithy-aws-protocol-tests/model/ec2Query/input.smithy]: I suggest removing the capitalization on `XmlNameOnly` and `QueryAndXmlName`. That would allow this to test the capitalization of xmlName and not-capitalizing ec2QueryName.
